homeassistant/components/ipp/sensor.py

```diff
@@ -80,7 +80,9 @@ def _get_marker_value_fn(
         device_class=SensorDeviceClass.TIMESTAMP,
         entity_category=EntityCategory.DIAGNOSTIC,
         entity_registry_enabled_default=False,
-        value_fn=lambda printer: (utcnow() - timedelta(seconds=printer.info.uptime)),
+        value_fn=lambda printer: (
+            utcnow() - timedelta(minutes=int(printer.info.uptime / 60))
+        ),
```

Comment:
Why don't we just calculate it only once in the coordinator at the first fetch and only update if a new value is smaller than the one we had (+ 60 seconds to make sure it can fluctuate)?

Reply:
This was the easier solution without put to much efforts into further changes of the coordinator, since the `coordinator.data` is just the `Printer` class from the pyipp lib. When we want to add further values into the coordinator data, we need put a bit more efforts.

Reply:
Further I think we would not safe anything, when checking (_calculating_) in the coordinator, if the new timestamp is "lower" than the previous one - we still need to calculate with timestamps :thinking:

Reply:
That can just be calculated with numbers
```py
if self.previous_uptime + 60 < new_uptime:
    ...
```

Reply:
But since this is platinum, also curious what @ctalkington thinks of it :)

(I don't have that strong of an opinion, just curious)

Reply:
ah ... now I get it - you mean to compare the `printer.info.uptime` not the timestamp itself - yeah, this would safe the uptime timestamp calculation on each data fetch - but still needs additional efforts to change the coordinator data accordingly ... will check this ...

Reply:
here we go 381750c

Reply:
I'm wondering if I can update pyipp to help with this issue as I'm not a fan of overloading the coordinator for this oddity

Reply:
since the reported uptime is in "seconds since boot", so you somehow need to track the previous state and compare it while requesting the new state - not sure, if you've such state history available within the `pyipp` lib 🤔

Reply:
pyipp 0.17 adds `booted_at` to `Printer` it's only changed when uptime slides backwards from previous request

Reply:
I've reverted all previous changes, bumped pyipp to 0.17.0 and just made use of the new `printer.booted_at` property